miri-script/src/main.rs

```diff
@@ -0,0 +1,63 @@
+pub(crate) mod arg;
+mod commands;
+
+use std::io::Result;
+use std::path::Path;
+
+use clap::Parser;
+use path_macro::path;
+
+use arg::Subcommands;
+
+struct AutoConfig {
+    toolchain: bool,
+    fmt: bool,
+    clippy: bool,
+}
+
+impl Subcommands {
+    fn supports_auto_things(&self) -> bool {
+        use Subcommands::*;
+        match self {
+            // Early commands, that don't do auto-things and don't want the environment-altering things happening below.
+            Toolchain { .. }
+            | RustcPull { .. }
+            | RustcPush { .. }
+            | ManySeeds { .. }
+            | Bench { .. } => false,
+            Install { .. }
+            | Check { .. }
+            | Build { .. }
+            | Test { .. }
+            | Run { .. }
+            | RunDep { .. }
+            | Bless { .. }
+            | Fmt { .. }
+            | Clippy { .. }
+            | Cargo { .. } => true,
+        }
+    }
+    fn get_auto_config(&self, miri_dir: &Path) -> Option<AutoConfig> {
+        use std::env::VarError;
+        if let Err(VarError::NotPresent) = std::env::var("MIRI_AUTO_OPS") {
+            std::env::set_var("MIRI_AUTO_OPS", "42");
```

Comment:
This shouldn't be needed any more; the shell script did this because it recursively invoked itself to run the auto-ops, but we do no such nonsense here.